From d5d6383d86c78c914293038ed981d7c99be4547e Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Fri, 18 Mar 2016 06:05:14 -0700
Subject: [PATCH 1600/1767] Revert "UBUNTU: SAUCE: fixup: 20/23 locking issue
 around in __label_update"

BugLink: http://bugs.launchpad.net/bugs/1379535

This reverts commit bd0bf02f160e39b34a63a8ab35ab34fff74efb40.

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 security/apparmor/label.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/apparmor/label.c b/security/apparmor/label.c
index 0b3a80a..ee9a573 100644
--- a/security/apparmor/label.c
+++ b/security/apparmor/label.c
@@ -1844,7 +1844,7 @@ static struct aa_label *__label_update(struct aa_label *label)
 	if (invcount) {
 		l->size -= aa_sort_and_merge_profiles(l->size, &l->ent[0]);
 		if (labels_set(label) != labels_set(l)) {
-			__aa_label_remove(labels_set(label), label, l);
+			aa_label_remove(labels_set(label), label);
 			write_unlock_irqrestore(&ls->lock, flags);
 			tmp = aa_label_insert(labels_set(l), l);
 			goto out;
-- 
1.9.3

